sqlite_query.c

```diff
@@ -325,3 +324,29 @@ static const char* sqlite_datatype(int t)
 			return azType[0];
 	}
 }
+
+/*
+ * Human readable message about disallowed combination of PostgreSQL columnn
+ * data type and SQLite data value affinity
+ */
+static void sqlite_value_to_pg_error (Oid pgtyp, int pgtypmod, sqlite3_stmt * stmt, AttrNumber attnum, int sqlite_value_affinity, int affinity_for_pg_column, int value_byte_size_blob_or_utf8)
+{
+	const char	*sqlite_affinity = 0;
+	const char	*pg_eqv_affinity = 0;
+	const char	*pg_dataTypeName = 0;
+	const int	 max_logged_byte_length = 63;
+	
+	pg_dataTypeName = TypeNameToString(makeTypeNameFromOid(pgtyp, pgtypmod));
+	sqlite_affinity = sqlite_datatype(sqlite_value_affinity);
+	pg_eqv_affinity = sqlite_datatype(affinity_for_pg_column);
+	
+	if (value_byte_size_blob_or_utf8 < max_logged_byte_length)
+	{
+		const unsigned char	*text_value = sqlite3_column_text(stmt, attnum);
+		elog(ERROR, "SQLite data affinity = \"%s\" disallowed for PostgreSQL type \"%s\" = SQLite \"%s\", value ='%s'", sqlite_affinity, pg_dataTypeName, pg_eqv_affinity, text_value);
+	}
+	else
+	{
+		elog(ERROR, "SQLite data affinity = \"%s\" disallowed for PostgreSQL type \"%s\" = SQLite \"%s\" for a long value", sqlite_affinity, pg_dataTypeName, pg_eqv_affinity);
```

Comment:
I know that in this case, the column value is too long.
Could you add the value length or part of the column value?

Reply:
> Could you add the value length or part of the column value?

Yes. Value length is better, because in case of 1,5 Gb `BLOB` our `sqlite3_column_text` will be memory eater. Unfortunately I don't know how to write substring and getter in one line. `sqlite3_column_blob` and `sqlite3_column_blob` gives only original long value.

Reply:
> Unfortunately I don't know how to write substring and getter in one line.

I don't know that also. 
I think the data length is enough. I confirmed the change.